subprojects/core/src/main/java/org/gradle/api/internal/file/copy/NormalizingCopyActionDecorator.java

```diff
@@ -53,31 +53,26 @@ public NormalizingCopyActionDecorator(CopyAction delegate, Chmod chmod) {
 
     @Override
     public WorkResult execute(final CopyActionProcessingStream stream) {
-        final Set<RelativePath> visitedDirs = new HashSet<RelativePath>();
+        final Set<RelativePath> visitedDirs = new HashSet<>();
         final ListMultimap<RelativePath, FileCopyDetailsInternal> pendingDirs = ArrayListMultimap.create();
 
-        WorkResult result = delegate.execute(new CopyActionProcessingStream() {
+        return delegate.execute(new CopyActionProcessingStream() {
```

Comment:
Nit: make this into a lambda (by moving the private method to the enclosing class)?

Reply:
Done.